### ISSUE TITLE
bug(products): The price first added to a variant is locked to USD

### ISSUE THREAD

**Issue by olavurdj** [medusajs/admin#525]
If I create a product variant on an existing product (or draft), the first price added is always locked to USD, even if USD is not configured as a currency. The problem persists after restart of the Admin panel after currency configuration (see below).

## Prerequisites
Make sure USD isn't configured as the default or store currency. On the Admin panel, go to **Settings** > **Currency**, and configure a non-USD currency (e.g. DKK) as both the **Store currencies** and **Default store currency**.

## Steps to reproduce
On the Admin panel: 
1. Go to **Products** > **New product**
2. Fill in a Name
3. Click **Save as draft**. 
4. Open the new product draft. 
5. On the **Variants** panel, click the three dots and then **Add variant**. 
6. On the **Add variant** modal, click **Add a price**

## Actual behaviour
The added price is locked to USD

## Expected behaviour
The added price to be in DKK (or the configured default currency/any store currency).

----
Medusa v1.2.1
Medusa Admin, 8189ab1c6b5cf2f92a226da30f7d665e7c8b39f9

Mozilla Firefox v99.0.1

**Comment by olivermrbl** [comment]
Thanks for reporting the bug. We are using `usd` as default currency code when initializing adding new prices to a variant. 

Instead we should use a currency present in the store, as you point out. 

A way to solve it would be to include options when [initializing the `usePricesFieldArray`](https://github.com/medusajs/admin/blob/8189ab1c6b5cf2f92a226da30f7d665e7c8b39f9/src/domain/products/details/variants/variant-editor.tsx#L51) like so:
```
const {
    fields: prices,
    appendPrice,
    deletePrice,
    availableCurrencies,
  } = usePricesFieldArray(
    store?.currencies.map((c) => c.code) || [],
    {
      control,
      name: "prices",
      keyName: "indexId",
    },
    {
      defaultAmount: 1000,
      defaultCurrencyCode: store?.currencies[0]?.code || "usd", // <- specifically this line solves it
    }
  )
```

Feel free to open a PR, if you are up for it! @olavurdj

**Comment by olivermrbl** [comment]
Solved in #526
